File: js210/js211_assessment/js210sp_easy3/letter_swap.js
"use strict";

function swap(text) {
  let words = text.split(' ');
  let modifiedWords = [];

  words.forEach(function (word) {
    modifiedWords.push(letterSwapped(word));
  });

  return modifiedWords.join(' ');
}


function letterSwapped(word) {
  let chars = word.split('');
  let firstChar = chars[0];
  let lastChar = chars[chars.length - 1];

  chars.splice(0, 1, lastChar);
  chars.splice(chars.length - 1, 1, firstChar);
  return chars.join('');
}

console.log(swap('Oh what a wonderful day it is'));
console.log(swap('Abcde'));
console.log('a');

File: rb109/Practice/Easy 5/letter_swap.rb
# letter_swap.rb

# write a method
# takes a string of words separated by spaces as input
# outputs a string with first and last letter of each word swapped

# split the string at spaces for each word
# for each word I need: first_char = word[0], last_char = word[-1]
# then, I need to do word[0] = last_char, and word[-1] = first_char

# array.each do |word|
#   first_char = word[0]
#   last_char = word[-1]
#   word[0] = last_char
#   word[-1] = first_char
# end

def swap(string)
  swapped = string.split.each do |word|
    first_char = word[0]
    last_char = word[-1]
    word[0] = last_char
    word[-1] = first_char
  end
  swapped.join(' ')
end

p swap('Oh what a wonderful day it is') == 'hO thaw a londerfuw yad ti si'
p swap('Abcde') == 'ebcdA'
p swap('a') == 'a'
t